### PR TITLE
Convert osu.groupColour to typescript

### PR DESCRIPTION
**Pull request by notbakaneko**

### CODE REVIEW

File: resources/assets/lib/utils/css.ts [hunk 1]
```diff
@@ -1,8 +1,15 @@
 // Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
 // See the LICENCE file in the repository root for full licence text.
 
+import GroupJson from 'interfaces/group-json';
 import { forEach } from 'lodash';
 
+declare module 'react' {
```
[the remark this answers is not in the document]
Comment (nanaya) [reply resources/assets/lib/utils/css.ts:7]: also I wonder if `` `--${string}` `` works...

[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/assets/lib/utils/css.ts:7]: It needs to be declared a type for the interface to extend but then extended interface definition is empty which makes an error that needs to be ignored 🤔

[the remark this answers is not in the document]
Comment (notbakaneko) [reply resources/assets/lib/utils/css.ts:7]: putting it in a `.d.ts` also requires an `import` to not override the existing react declarations, but that also triggers a `no-unused-vars` error....

[the remark this answers is not in the document]
Comment (nanaya) [reply resources/assets/lib/utils/css.ts:7]: apparently just `import 'react'` works? 🙃

**Comment by nanaya** [comment]
conflicts

**Review by nanaya — APPROVED** [review]

**Pull request merged**